doc/OnlineDocs/tutorial/index.rst

```diff
@@ -29,6 +29,72 @@ Concrete vs Abstract Models
 
 Modeling Components
 -------------------
+The primary modeling components defined by Pyomo are
+:class:`Set<pyomo.core.base.sets.Set>`, :class:`Var<pyomo.core.base.var.Var>`,
+:class:`Constraint<pyomo.core.base.constraint.Constraint>`,
+:class:`Objective<pyomo.core.base.objective.Objective>`, and
+:class:`Block<pyomo.core.base.block.Block>`. Each of these may be indexed.
+
+.. doctest::
+   
+   >>> import pyomo.environ as pe
+   >>> m = pe.ConcreteModel()
+   >>> m.a = pe.Set(initialize=[1, 2, 3])
+   >>> m.x = pe.Var(m.a, initialize=0, bounds=(-10,10))
+   >>> m.y = pe.Var(m.a)
+   >>> def c_rule(m, i):
+   ...     return m.x[i] >= m.y[i]
+   >>> m.c = pe.Constraint(m.a, rule=c_rule)
+   >>> m.c.pprint()
+   c : Size=3, Index=a, Active=True
+       Key : Lower : Body        : Upper : Active
+         1 :  -Inf : y[1] - x[1] :   0.0 :   True
+	 2 :  -Inf : y[2] - x[2] :   0.0 :   True
+	 3 :  -Inf : y[3] - x[3] :   0.0 :   True
+
+By default, the index set is fixed. New indices may not be added:
+
+.. doctest::
+   
+   >>> m.z = pe.Var()
+   >>> m.c[4] = m.x[1] == 5 * m.z
+   Traceback (most recent call last):
+     ...
+   KeyError: "Index '4' is not valid for indexed component 'c'"
+
+This helps prevent many common mistakes. However, IndexedComponent can behave
+more like a dictionary if Any is used as the index.
```

Comment:
I think we should document IndexComponent and Any as classes:   :class:`IndexComponent`  and :class:`Any`.